src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12575,7 +12576,15 @@ void Unit::_ExitVehicle(Position const* exitPosition)
         // Set exit position to vehicle position and use the current orientation
         pos = vehicle->GetBase()->GetPosition();
         pos.SetOrientation(GetOrientation());
+
+        // To-do: snap this hook out of existance
         sScriptMgr->ModifyVehiclePassengerExitPos(this, vehicle, pos);
```

Comment:
I don't think, that this hook is necesary to remove, could exist some vehicle where the exit position could be changed in a dynamic way.

Reply:
That hook, or better said all scripts of the scriptmgr that use the same method are absolutely just custom script features. This hook is being slapped on every single vehicle. The more scripts use it, the more scripts will be called with every single vehicle that's ejecting a passenger. That's absolutely not okay and should get dropped as soon as possible. We have a proper way now.